### PR TITLE
DDCNLS-7127 - Added client ID and session ID to all pages

### PR DESCRIPTION
**Pull request by brendanstraughan**

### CODE REVIEW

File: app/views/main.scala.html [hunk 1]
```diff
@@ -55,6 +55,13 @@
         seq
     }
 
+@gaCustomEvent = @{
```
Comment (oc-ste) [inline app/views/main.scala.html]: gaCustomEvent is never used, consider adding it to inlineScript

File: app/views/main.scala.html [hunk 2]
```diff
@@ -67,47 +75,34 @@
...
+
+<script>
                 window.GOVUK.timeoutDialog({
                     timeout: @configDecorator.sessionTimeoutInSeconds,
                     countdown: @configDecorator.sessionCountdownInSeconds,
                     keepAliveUrl: '@controllers.routes.SessionManagementController.keepAlive()',
                     signOutUrl: '@controllers.routes.SessionManagementController.timeOut()',
                 });
         </script>
+@if(displayTimeoutDialog) {
```
Comment (oc-ste) [inline app/views/main.scala.html]: This should be wrapped around:

`            window.GOVUK.timeoutDialog({	                window.GOVUK.timeoutDialog({
                    timeout: @configDecorator.sessionTimeoutInSeconds,	                    timeout: @configDecorator.sessionTimeoutInSeconds,
                    countdown: @configDecorator.sessionCountdownInSeconds,	                    countdown: @configDecorator.sessionCountdownInSeconds,
                    keepAliveUrl: '@controllers.routes.SessionManagementController.keepAlive()',	                    keepAliveUrl: '@controllers.routes.SessionManagementController.keepAlive()',
                    signOutUrl: '@controllers.routes.SessionManagementController.timeOut()',	                    signOutUrl: '@controllers.routes.SessionManagementController.timeOut()',
                });`

File: conf/application.conf [hunk 3]
```diff
@@ -53,7 +53,7 @@ controllers {
   }
 }
 
-play.filters.headers.contentSecurityPolicy = "default-src 'self' 'unsafe-inline' 'unsafe-eval' localhost:9032 localhost:9250 localhost:9310 www.google-analytics.com assets.publishing.service.gov.uk localhost:8291 *.analytics-egain.com stats.g.doubleclick.net www.google-analytics.com data:"
+play.filters.headers.contentSecurityPolicy = "default-src 'self' 'unsafe-inline' 'unsafe-eval' localhost:9032 localhost:9250 localhost:9310 www.google-analytics.com assets.publishing.service.gov.uk localhost:8291 *.analytics-egain.com stats.g.doubleclick.net www.google-analytics.com *.optimizely.com data:"
```
Comment (oc-ste) [inline conf/application.conf]: We need to ensure that the CSP is updated in other environments to allow optimizely through, feel free to remove this if you want and create a new ticket to allow Optimizely through CSP

File: app/views/google_analytics.scala.html [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+@(sessionId: String, langCode: String, url: String)
+
+        ga('set', 'dimension5', '@langCode');
+        ga(function(tracker) {
+        var clientId = tracker.get('clientId');
+        ga('set', 'dimension66', clientId);
+        });
+        ga('set', 'page', '@url');
+        ga('set', 'anonymizeIp', true);
```
Comment (oc-ste) [inline app/views/google_analytics.scala.html]: You can remove this as it's in the main template

File: app/views/main.scala.html (hunk 1 above)
Comment (brendanstraughan) [reply app/views/main.scala.html]: done

File: app/views/google_analytics.scala.html (hunk 4 above)
Comment (brendanstraughan) [reply app/views/google_analytics.scala.html]: done

File: app/views/main.scala.html (hunk 2 above)
Comment (brendanstraughan) [reply app/views/main.scala.html]: done

File: app/views/google_analytics.scala.html (hunk 4 above)
Comment (brendanstraughan) [reply app/views/google_analytics.scala.html]: done

File: conf/application.conf (hunk 3 above)
Comment (brendanstraughan) [reply conf/application.conf]: will create a different ticket

Comment (brendanstraughan) [reply conf/application.conf]: https://jira.tools.tax.service.gov.uk/browse/DDCNLS-7157

Have also linked this to the ticket DDCNLS-7127

File: app/views/main.scala.html [hunk 5]
```diff
@@ -67,36 +75,23 @@
...
-                    }));
-                };
-        </script>
-    }
 
-    @if(displayTimeoutDialog) {
+@scriptElement
+
+@if(configDecorator.analyticsToken == Some("N/A")) {
+    }
```
Comment (RobertLaverick) [inline app/views/main.scala.html]: Is this meant to be an empty if? Unless I'm misreading it, this looks like it isn't doing anything. Can you confirm? @brendanstraughan

**Comment by RobertLaverick** [comment]
:shipit:

**Pull request merged**